UT4MasterServer/Models/StatisticBase.cs

```diff
@@ -1110,4 +779,176 @@ public List<string> Validate()
+		ValidatePerWeaponKillStatistic(BioRifleKills, nameof(BioRifleKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(BioLauncherKills, nameof(BioLauncherKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(ShockBeamKills, nameof(ShockBeamKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(ShockCoreKills, nameof(ShockCoreKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(ShockComboKills, nameof(ShockComboKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(LinkKills, nameof(LinkKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(LinkBeamKills, nameof(LinkBeamKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(MinigunKills, nameof(MinigunKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(MinigunShardKills, nameof(MinigunShardKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(FlakShardKills, nameof(FlakShardKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(FlakShellKills, nameof(FlakShellKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(RocketKills, nameof(RocketKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(SniperKills, nameof(SniperKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(SniperHeadshotKills, nameof(SniperHeadshotKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(LightningRiflePrimaryKills, nameof(LightningRiflePrimaryKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(LightningRifleSecondaryKills, nameof(LightningRifleSecondaryKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(RedeemerKills, nameof(RedeemerKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(InstagibKills, nameof(InstagibKills), flaggedFields);
+		ValidatePerWeaponKillStatistic(TelefragKills, nameof(TelefragKills), flaggedFields);
+		#endregion
+
+		return flaggedFields;
+	}
+
+	private static void ValidateMaxValueForStatistic(int? value, int maxValue, string statisticName, List<string> flaggedFields)
+	{
+		if (value.HasValue && value.Value > maxValue)
+		{
+			flaggedFields.Add(statisticName);
+		}
+	}
+
+	private static void ValidateIfBothStatisticsExist(int? firstValue, int? secondValue, string firstStatistic, string secondStatistic, List<string> flaggedFields)
+	{
+		if (firstValue.HasValue && secondValue.HasValue && firstValue.Value > 0 && secondValue.Value > 0)
+		{
+			flaggedFields.Add(firstStatistic);
+			flaggedFields.Add(secondStatistic);
+		}
+	}
+
+	private void ValidateStatisticsAgainstTimePlayed(int? value, double maxValuePerSecond, string statisticName, List<string> flaggedFields)
+	{
+		if (value.HasValue && TimePlayed.HasValue && value.Value > (TimePlayed.Value / maxValuePerSecond))
+		{
+			flaggedFields.Add(statisticName);
+			flaggedFields.Add(nameof(TimePlayed));
+		}
+	}
+
+	private void ValidateMultiAndSpreeKillStatistic(int? value, int multiplier, string statisticName, List<string> flaggedFields)
+	{
+		if (value.HasValue)
+		{
+			if (!Kills.HasValue || (Kills.HasValue && value.Value * multiplier > Kills.Value))
```

Comment:
Merge it into one IF statement or return like this to reduce nesting:

```C#
if (!value.HasValue)
{
	return;
}
```

Reply:
Done.